shared/src/main/scala/scodec/Attempt.scala

```diff
@@ -75,6 +80,16 @@ object Attempt {
   /** Creates an unsuccessful attempt. */
   def failure[A](err: Err): Attempt[A] = Failure(err)
 
+  /** Creates a successful attempt if the condition succeeds otherwise create a unsuccessful attempt */
+  def guard(condition: => Boolean, err: String) =
```

Comment:
In most cases if you would use this combinator, you wont need to add context to the error. I usually use this for CRC checks or size checks etc. Though I agree that it's better to `Err` as well. Maybe create a overloaded version? So error as either `String` or `Err` ?